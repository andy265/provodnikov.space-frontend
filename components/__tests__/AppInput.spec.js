const { createRenderer } = require('vue-server-renderer')
const { mount, shallow } = require('vue-test-utils')
const AppInput = require('~/components/AppInput')

describe('AppInput.vue', () => {
  it('has same HTML structure', () => {
    let wrapper = mount({
      template: '<AppInput value="some text" placeholder="Text"/>',
      components: {
        AppInput
      }
    })
    let renderer = createRenderer()
    // eslint-disable-next-line
    renderer.renderToString(wrapper.vm, (err, str) => {
      expect(str).toMatchSnapshot()
    })
  })

  it('emit input event on change', () => {
    let wrapper = shallow(AppInput, {
      propsData: {
        value: 'some text'
      }
    })
    wrapper.find('input').element.value = 'new text'
    wrapper.find('input').trigger('input')
    expect(wrapper.emitted().input[0][0]).toEqual('new text')
  })
})
