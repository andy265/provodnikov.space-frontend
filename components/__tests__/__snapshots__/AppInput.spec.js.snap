// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppInput.vue has same HTML structure 1`] = `<input type="text" placeholder="Text" value="some text" class="app-input">`;
